endoreg_db/views/video/video_stream.py

```
@@ -0,0 +1,306 @@
+            try:
+                # Open file in binary mode - FileResponse will handle closing
+                file_handle = open(path, 'rb')
+                response = FileResponse(file_handle, content_type=content_type)
+                
+                # Set HTTP headers for video streaming
+                response['Content-Length'] = str(file_size)
+                response['Accept-Ranges'] = 'bytes'  # Enable HTTP range requests for seeking
+                response['Content-Disposition'] = f'inline; filename="{path.name}"'
+                
+            except IOError as e:
+                raise Http404(f"Cannot open video file: {str(e)}")
```

Ruff (0.14.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)